x-pack/plugins/cases/server/routes/api/register_routes.ts

```diff
@@ -0,0 +1,84 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License
+ * 2.0; you may not use this file except in compliance with the Elastic License
+ * 2.0.
+ */
+
+import { schema } from '@kbn/config-schema';
+import { RouteRegistrar } from 'kibana/server';
+import { CasesRequestHandlerContext } from '../../types';
+import { CaseRoute, RegisterRoutesDeps } from './types';
+import { escapeHatch, getIsKibanaRequest, wrapError } from './utils';
+
+const increaseTelemetryCounters = ({
+  telemetryUsageCounter,
+  method,
+  path,
+  isKibanaRequest,
+  isError = false,
+}: {
+  telemetryUsageCounter: Exclude<RegisterRoutesDeps['telemetryUsageCounter'], undefined>;
+  method: string;
+  path: string;
+  isKibanaRequest: boolean;
+  isError?: boolean;
+}) => {
+  const counterName = `${method.toUpperCase()} ${path}`;
+
+  telemetryUsageCounter.incrementCounter({
+    counterName,
+    counterType: isError ? 'error' : 'success',
```

Comment:
I am wondering if it will be useful to have a different counter type for `deprecation` like `deprecation error` or `deprecation success`

Reply:
Nice suggestion! I plan to move the deprecation login inside the central handler in another PR. I will add the deprecation counters in the new PR.